mmdet/datasets/pipelines/translate.py

```diff
@@ -0,0 +1,240 @@
+            range 0 to 1.
+        fill_val (int | float | tuple): The filled values for image border.
+            If float, the same fill_val will be used for all the three channels
+            of image. If tuple, the should be 3 elements (e.g. equals the
+            number of channels for image).
+        seg_ignore_label (int): The ``fill_val`` used for segmentation map.
+            Note this value must equals ``ignore_label`` in ``semantic_head``
+            of the corresponding config. Default 255.
+        axis (str): Translate images along with x-axis or y-axis. The option
+            of axis is 'x' or 'y'.
+        max_translate_offset (int | float): Pixel's maximum offset value for
+            Translate along with x-axis or y-axis.
+    """
+
+    def __init__(self,
+                 level,
+                 prob=0.5,
+                 fill_val=128,
+                 seg_ignore_label=255,
+                 axis='x',
+                 max_translate_offset=250.,
+                 *args,
+                 **kwargs):
+        assert isinstance(level, (int, float)), \
+            'The level must be type int or float.'
+        assert 0 <= level <= _MAX_LEVEL, \
+            'The level used for calculating Translate\'s offset should be ' \
+            'in range (0,_MAX_LEVEL]'
+        assert 0 <= prob <= 1.0, \
+            'The probability of translation should be in range 0 to 1.'
+        if isinstance(fill_val, (float, int)):
+            fill_val = tuple([float(fill_val)])
+        elif isinstance(fill_val, tuple):
+            assert len(fill_val) == 3, \
+                'fill_val as tuple must have 3 elements.'
+            fill_val = tuple([float(val) for val in fill_val])
+        else:
+            raise ValueError(
+                'fill_val must be a float scale or a tuple with 3 elements.')
+        assert np.all([0 <= val <= 255 for val in fill_val]), \
+            'all elements of fill_val should between range [0,255].'
+        assert axis in ('x', 'y'), \
+            'Translate should be alone with x-axis or y-axis.'
+        assert isinstance(max_translate_offset, (int, float)), \
+            'The max_translate_offset must be type int or float.'
+        # the offset for translation
+        self.offset = int(level_to_value(level, max_translate_offset))
+        self.level = level
+        self.prob = prob
+        self.fill_val = fill_val
+        self.seg_ignore_label = seg_ignore_label
+        self.axis = axis
+        self.max_translate_offset = max_translate_offset
+
+    @staticmethod
+    def get_translate_matrix(offset, axis='x'):
+        """Generates the transformation matrix."""
+        if axis == 'x':
+            trans_matrix = np.float32([[1, 0, offset], [0, 1, 0]])
+        elif axis == 'y':
+            trans_matrix = np.float32([[1, 0, 0], [0, 1, offset]])
+        return trans_matrix
+
+    @staticmethod
+    def warpAffine(data,
```

Comment:
`warpAffine` in `mask_structure` is duplicated with this function here. It may be better to call  `cv2.warpAffine` to simplify and prevent duplication.

Reply:
Hi, I have removed the duplicated warpAffine function for both translate.py and structures.py, now cv2.warpAffine is directly used.